app/lib/env.server.ts

```diff
@@ -44,6 +44,22 @@ function getHostname() {
   return new URL(getOrigin()).hostname
 }
 
+function getDomain() {
+  const hostname = getHostname() as
+    | 'gcn.nasa.gov'
+    | 'test.gcn.nasa.gov'
+    | 'dev.gcn.nasa.gov'
+    | undefined
```

Comment:
getHostname is not constrained to return one of those values.

```suggestion
  const hostname = getHostname()
```

Reply:
True, but they are the only valid values for domain, so should this cast be done elsewhere?

Reply:
For example, in kafka.server.ts:

```js
const producer = new Kafka({
  client_id,
  client_secret,
  domain: domain as
    | 'gcn.nasa.gov'
    | 'test.gcn.nasa.gov'
    | 'dev.gcn.nasa.gov'
    | undefined,
}).producer()
```

Reply:
> True, but they are the only valid values for domain, so should this cast be done elsewhere?

They are the _only valid return values from this function_. However, `getHostname()` can return any string, or `undefined`. So this cast is incorrect.

Reply:
See #2295.